src/plugins/newsfeed/public/lib/convert_items.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import moment from 'moment';
+import { ApiItem, NewsfeedItem } from '../types';
+import { NEWSFEED_FALLBACK_LANGUAGE } from '../../common/constants';
+
+export const convertItems = (items: ApiItem[], userLanguage: string): NewsfeedItem[] => {
+  return items
+    .filter(validatePublishedDate)
+    .map((item) => convertItem(item, userLanguage))
+    .filter(validateIntegrity);
+};
+
+export const validatePublishedDate = (item: ApiItem): boolean => {
+  if (moment(item.expire_on).isBefore(Date.now())) {
+    return false; // ignore item if expired
+  }
+
+  if (moment(item.publish_on).isAfter(Date.now())) {
+    return false; // ignore item if publish date hasn't occurred yet (pre-published)
+  }
+  return true;
+};
+
+export const convertItem = (rawItem: ApiItem, userLanguage: string): NewsfeedItem => {
```

Comment:
nit: seems like `convert` is a pretty vague term here, maybe this function and file should be `localize_items` instead?

Reply:
it makes sense to rename `convertItem` to `localizeItem`, however `convertItems`  does more than that, as it also validates the content/dates of the news, so not sure the whole file should be renamed?

Reply:
Fine with me 👍